### PR TITLE
XPath injection kb entry

### PR DESCRIPTION
**Pull request by ErebusZ**
XPath injection kb entry.

### CODE REVIEW

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/description.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.StringReader
+
+fun main() {
+    val userInput = readLine() ?: ""
+    val factory = DocumentBuilderFactory.newInstance()
+    val builder = factory.newDocumentBuilder()
+    val inputSource = InputSource(StringReader(xml))
+    val document = builder.parse(inputSource)
+    val name = document.getElementsByTagName("name").item(0).textContent
+    println("Hello, $name!")
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/description.md:60]: Where is the XPath vulnerability here?

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/meta.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		"OWASP_MASVS_L2": [
+            "MSTG_PLATFORM_2"
+		],
+        "CWE_TOP_25": [
+            "CWE_89"
+        ],
+        "OWASP_TOP_10": [
+			"A03:2021-Injection"
+		]
+    }
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/meta.json:24]: The formatting of this seems off.

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+
+# Xpath Injection
+
+To mitigate XPath Injection vulnerabilities, it is important to properly validate and sanitize user input before using it in XPath queries. This can be done by using parameterized queries or prepared statements, which separate the user input from the query logic. Additionally, limiting the privileges of the user executing the query can help prevent unauthorized access to sensitive data. Regularly updating and patching software and libraries can also help prevent known vulnerabilities from being exploited. Finally, implementing logging and monitoring can help detect and respond to any attempted attacks.
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md:4]: There parametrized queries on xpath?

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+### Kotlin
+
+```kotlin
+import javax.xml.parsers.DocumentBuilderFactory
+import org.xml.sax.InputSource
+import java.io.StringReader
+
+fun main() {
+    val userInput = readLine() ?: ""
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md:74]: I don't see the issue here.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/meta.json (hunk 2 above)
Comment (ErebusZ) [reply MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/meta.json]: [0285859](https://github.com/Ostorlab/KB/pull/58/commits/0285859eaa66eb530bf095a3e3bbf4711eeab044)

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md (hunk 3 above)
Comment (ErebusZ) [reply MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md:4]: depends on the package but yes, as far as I understand

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/description.md (hunk 1 above)
Comment (ErebusZ) [reply MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/description.md]: [3e35e7e](https://github.com/Ostorlab/KB/pull/58/commits/3e35e7e983b4e854c4e61952b620b54b4fa82837)

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md (hunk 4 above)
Comment (ErebusZ) [reply MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md]: [d270584](https://github.com/Ostorlab/KB/pull/58/commits/d27058446edcfd6aedc081445ea756617e2dec5a), please double check the new snippet, also this one uses parameterized query.

File: MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                <username>Bob</username>
+                <password>pass456</password>
+            </user>
+        </users>
+    """.trimIndent()
+
+    val xpathQuery = "//*[username/text()='${userInput}']"
+
+    val xpath = XPathFactory.newInstance().newXPath()
+    val expression = xpath.compile(xpathQuery)
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md:88]: Isn't this still vulnerable?

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**